js/models/conversations.js

```diff
@@ -72,6 +72,12 @@
         this.initialPromise = Promise.resolve();
 
         this.contactCollection = new Backbone.Collection();
+        var collator = new Intl.Collator();
+        this.contactCollection.comparator = function(left, right) {
+            left = left.getTitle().toLowerCase();
```

Comment:
this is a potential problem for future.
I would not change reference of any argument variable, for just comparison .
why not use 
`var leftTitle = left.getTitle().toLowerCase();`

just to be on a safe side. hm?

Reply:
What are you worried about, exactly? It's not changing the incoming data. It's just rebinding the value of the `left` and `right` variables.

Reply:
left and right  are objects in this case, that might have references pointing on them.
So changing reference of left object to some string, is not a good idea, and will lead to potentially unsafe an unpredictable code, somewhere else in the program. IMHO..

Reply:
You might find some value in this article about by-value vs. by-ref in Javascript: https://stackoverflow.com/questions/6605640/javascript-by-reference-vs-by-value

Reply:
thanks a lot. that was very useful.
actually I agree, you change binding of left to point to some string.

My be, I am used to code in such a way, no questions are left, so I wouldn't do it this way, but anyway it's OK. :) thanks again.

I wanted to contribute to code, if you can advice me a simple thing to start from, I would be happy.

Reply:
You might look through the items in the Backlog column here, grab one of the simpler ones... https://github.com/WhisperSystems/Signal-Desktop/projects/1